src/_pytest/pytester.py

```diff
@@ -614,7 +614,7 @@ def to_text(s):
 
         ret = None
         for basename, value in items:
-            p = self.tmpdir.join(basename).new(ext=ext)
+            p = self.tmpdir.join(basename, abs=True).new(ext=ext)
```

Comment:
this is a breaking change that opens up badly made test suites to accidentally writing system wide files

the pathlib behaviour is a security nightmare as is

Reply:
@RonnyPfannschmidt 
So what changes do you request then actually?
The use case here is using `testdir.makepyfile` with an absolute file - do you think this should not be supported then?

Reply:
Testing the current behavior in https://github.com/pytest-dev/pytest/pull/6579.

Reply:
The current behavior fails on Windows:

> OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw1\\test_makefile_joins_absolute_path0\\C:'